apps/vs-code-designer/src/app/utils/binaries.ts

```diff
@@ -108,15 +110,15 @@ export async function downloadAndExtractDependency(
           await extractDependency(dependencyFilePath, targetFolder, dependencyName);
           vscode.window.showInformationMessage(localize('successInstall', `Successfully installed ${dependencyName}`));
         }
-        resolve();
       });
-      downloadStream.on('error', reject);
+      writer.on('error', async (error) => {
+        throw error;
+      });
     });
   } catch (error) {
     vscode.window.showErrorMessage(`Error downloading and extracting the ${dependencyName} zip file: ${error.message}`);
     await executeCommand(ext.outputChannel, undefined, 'echo', `[ExtractError]: Remove ${targetFolder}`);
```

Comment:
Do you think it might be worth it to log some telemetry on the error?

Reply:
added context parameter, and logging telemetry